### PR TITLE
fix: Add page from template custom js lib is not saved

### LINKED ISSUE #31372: Bug: Add a page from template, does not add custom js libs
[Video showing add a template failing](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNXVjQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--e94a9406a366c097dc46632f0d8a6774969658d4/Screen%20Recording%202024-02-12%20at%201.57.05%E2%80%AFPM.mov)

- [x] Fix custom js libs
- [x] Check if there is anything else we can do to add this.

### PR DESCRIPTION
**Pull request by AnaghHegde**
## Description
When a page is added from template using the new page via template CTA, the customjs lib used is not saved. This is because of the condition used in Import Export flow for append to application flow. 

#### PR fixes following issue(s)
Fixes #31372 

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

## Testing
- [ ] Manual
- [ ] JUnit

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
	- Enhanced import functionality to better handle Custom JavaScript Libraries, ensuring smoother re-imports and updates.
- **Tests**
	- Added new tests to verify the improved handling and integrity of Custom JavaScript Libraries during the import process.

### CODE REVIEW

File: app/server/appsmith-server/src/test/java/com/appsmith/server/imports/internal/ImportServiceTests.java [hunk 1]
```diff
@@ -4286,6 +4286,11 @@ public void mergeApplication_nonGitConnectedApplicationSelectedSpecificPages_sel
                     List<NewAction> actionList = tuple.getT3();
                     List<ActionCollection> actionCollectionList = tuple.getT4();
 
+                    assertThat(application1.getUnpublishedCustomJSLibs().size()).isEqualTo(2);
+                    application1.getUnpublishedCustomJSLibs().forEach(customJSLib -> {
+                        assertThat(customJSLib.getUidString()).containsAnyOf("accessor1_url", "xmlParser");
```
Comment (NilanshBansal) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/imports/internal/ImportServiceTests.java:4291]: `containsAnyOf` will check for any of the value present for each element.

Comment (NilanshBansal) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/imports/internal/ImportServiceTests.java:4291]: @AnaghHegde `containsAnyOf` will check for any of the value present for each element. 
The test will also pass for the scenario when the customJSLib list has both the elements as ["accessor1_url", "accessor1_url"] but ideally we want to collect the names from list and then do comparison.

**Review by NilanshBansal — CHANGES_REQUESTED** [review]

**Review by NilanshBansal — DISMISSED** [review]
LGTM

**Review by sondermanish — APPROVED** [review]

**Pull request merged**